### PR TITLE
feat: remove unnecessary bundle layers and refactor verification

### PR DESCRIPTION
**Pull request by UncleGedd**
## Description
- ensure we only bundle images from required + selected components (results in smaller bundles)
- remove slow layer verification

## Related Issue
https://github.com/defenseunicorns/uds-cli/issues/613

### CODE REVIEW

File: src/test/e2e/optional_bundle_test.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+)
+
+func TestBundleOptionalComponents(t *testing.T) {
+	deployZarfInit(t)
+	e2e.SetupDockerRegistry(t, 888)
+	defer e2e.TeardownRegistry(t, 888)
+
+	// create 2 Zarf pkgs to be bundled
+	zarfPkgPath := "src/test/packages/podinfo-and-nginx"
+	e2e.CreateZarfPkg(t, zarfPkgPath, false)
```
Comment (decleaver) [inline src/test/e2e/optional_bundle_test.go:26]: consider making this force create? setting to true. These packages also get created in other tests I believe, which was causing this test to fail locally.

File: src/test/e2e/optional_bundle_test.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+	zarfPkgPath = "src/test/packages/prometheus"
+	pkg := filepath.Join(zarfPkgPath, fmt.Sprintf("zarf-package-prometheus-%s-0.0.1.tar.zst", e2e.Arch))
+	e2e.CreateZarfPkg(t, zarfPkgPath, false)
```
Comment (decleaver) [inline src/test/e2e/optional_bundle_test.go:30]: consider making this force create? setting to true. These packages also get created in other tests I believe, which was causing this test to fail locally.

File: src/test/e2e/optional_bundle_test.go (hunk 1 above)
Comment (UncleGedd) [reply src/test/e2e/optional_bundle_test.go:26]: good call!

File: src/test/e2e/optional_bundle_test.go (hunk 2 above)
Comment (UncleGedd) [reply src/test/e2e/optional_bundle_test.go:30]: good call!

File: src/test/e2e/optional_bundle_test.go (hunk 1 above)
Comment (decleaver) [reply src/test/e2e/optional_bundle_test.go:26]: disregard, should be a one and done, will only have to recreate this package if the zarf.yaml gets updated. So good as is.

File: src/test/e2e/optional_bundle_test.go (hunk 2 above)
Comment (decleaver) [reply src/test/e2e/optional_bundle_test.go:30]: same as previous comment. good as is

**Review by decleaver — APPROVED** [review]
Looks good! synchronous walk through, async review and local testing

**Pull request merged**